### DISCUSSION TITLE
[fs-extra] `writeFile`'s data type is too loose

### DISCUSSION THREAD

**Discussion by Josh-Cena** [DefinitelyTyped/DefinitelyTyped#59175] [Issues with a @types package]
`writeFile` and `outputFile` are typed with `data: any`:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/10ea84a4eb0b949af739c5837d5ade42438bc4ae/types/fs-extra/index.d.ts#L465-L476

However, `fs.writeFile("test.txt", { foo: "bar" });` panics:

```
TypeError [ERR_INVALID_ARG_TYPE]: The "data" argument must be of type string or an instance of Buffer, TypedArray, or DataView. Received an instance of Object
```

I think the type should be constrained to `string | NodeJS.ArrayBufferView`?

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
is this same issue? https://github.com/isaacs/node-graceful-fs/issues/211#issuecomment-1052531660
according to a comment, we should wait until fs-extra drops support for Node v12

**Reply by Josh-Cena** [reply]
Mmm, indeed, I see...
